torch/_inductor/kernel/flex_decoding.py

```diff
@@ -294,10 +297,12 @@ def flex_decoding_grid(batch_size, kv_heads, gqa_group_size, n_keys, d_model, me
 )
 
 
-def get_split_k(B: int, H: int, Mk: int, SM: int = 128) -> int:
-    """Heuristic for the number of splits from xformer"""
+def get_split_k(B: int, H: int, Mk: int) -> int:
+    num_SM = torch.cuda.get_device_properties("cuda").multi_processor_count
```

Comment:
I think we should cache this since it can be quite expensive, e.g. make a func and use lru_cache(none)